Dockerfile

```diff
@@ -6,31 +6,32 @@
 FROM alpine
 
 ARG GID=1000
+ENV GID=$GID
 ARG GNAME=defaultgroup
-ARG USER=defaultuser
-
-ENV HOME /home/$USER
+ENV GNAME=$GNAME
+ARG UNAME=defaultuser
+ENV UNAME=$UNAME
 
 # Install packages as root
-RUN apk upgrade --no-cache -a
-RUN apk update --no-cache --quiet
-RUN apk add --no-cache --quiet sudo curl
-
+RUN set -x &&\
+    apk add --update --upgrade curl doas &&\
+    apk upgrade &&\
+    rm -fr /var/cache/apk/* &&\
 # Add group
-RUN addgroup -g $GID $GNAME
+    addgroup -g $GID $GNAME &&\
 # Add new user
-RUN adduser -D $USER -G $GNAME \
-        && mkdir -p /etc/sudoers.d \
-        && echo "$USER ALL=(ALL) NOPASSWD: ALL" > /etc/sudoers.d/$USER \
-        && chmod 0440 /etc/sudoers.d/$USER
+    adduser -G $GNAME -D $UNAME &&\
+# Allow $UNAME to sudo/doas as root
+  echo "permit nopass $UNAME" >> /etc/doas.d/$UNAME.conf &&\
+  echo "permit nopass 0" >> /etc/doas.d/root.conf &&\
+  ln -s doas /usr/bin/sudo
 
-USER $USER
-WORKDIR $HOME
+USER $UNAME
+WORKDIR /home/$UNAME
 
-# Files in /home/$USER to be owned by $USER
-# Docker has --chown flag for COPY, but it does not expand ENV so we fallback to:
+# Files in /home/$UNAME to be owned by $UNAME
 # COPY src src
-# RUN sudo chown -R $USER:$USER $HOME
+# RUN sudo chown -R $UNAME:$GNAME /home/$UNAME
 
-CMD echo "User $(whoami) running from $PWD with premissions: $(sudo -l)"
+CMD echo "User $(whoami) running from $PWD with premissions: $(doas env)"
```

Comment:
Since 3.16 https://wiki.alpinelinux.org/wiki/Release_Notes_for_Alpine_3.16.0#doas-sudo-shim

Reply:
added `doas-sudo-shim` instead of symlink